src/main.rs

```diff
@@ -36,7 +36,11 @@ fn run() -> Fallible<()> {
     let config = JobConfig::from_file(&opt.jobconfig).context("reading job config")?;
 
     let mgr = SessionMPI::init(opt.hub)?;
-    println!("HOSTFILE:\n{}", mgr.hostfile()?);
+
+    if let Some(sources) = config.sources {
+        mgr.deploy(opt.jobconfig.parent().unwrap(), &sources)
```

Comment:
`unwrap` here seems dangerous. What if we run from a container such as Docker and have everything under `/`? Then `parent()` would return `None`.

Reply:
Should `opt.jobconfig.parent()` be `None`, it wouldn't be a valid file path, so we'd already have gotten an error in main.rs:36. 

Should I add a comment about it?

Reply:
Yeah, that'd be very helpful, thanks!

Reply:
Added.

Reply:
:+1: